graphx/src/main/scala/org/apache/spark/graphx/impl/GraphImpl.scala

```diff
@@ -178,6 +178,39 @@ class GraphImpl[VD: ClassTag, ED: ClassTag] protected (
     new GraphImpl(newVerts, replicatedVertexView.withEdges(newEdges))
   }
 
+  override def union[VD2: ClassTag, ED2: ClassTag, VD3: ClassTag, ED3: ClassTag] (
+    other: Graph[VD2, ED2],
+    mergeSameVertexAttr: (Option[VD], Option[VD2]) => VD3,
+    mergeSameEdgeAttr: (Option[ED], Option[ED2]) => ED3): Graph[VD3, ED3] = {
+
+    val newVertexRDD: RDD[(VertexId, VD3)] = vertices.fullOuterJoin(other.vertices).map {
+      pair => (pair._1, mergeSameVertexAttr(pair._2._1, pair._2._2))
+    }.cache()
+
+    //  convert other EdgeRDD to kv pair RDD
+    val otherPair = other.edges.mapPartitions {
+      iter => iter.map { edge => (edge.srcId.toString + edge.dstId.toString, edge) }
```

Comment:
I believe you should add some delimiter when concatenating. Otherwise two different edges can be glued together (i.e. 1 -> 12 and 11 -> 2).